overlord/configstate/configcore/certs.go

```diff
@@ -32,6 +33,221 @@ import (
 	"github.com/snapcore/snapd/overlord/configstate/config"
 )
 
+type certificate struct {
+	Name     string
+	Path     string
+	RealPath string
+}
+
+// This function validates the name of the certificate. We only allow certificates
+// with the correct suffix ".crt" except for the ca-certificates.crt,
+// or if the name is in the blockedCerts lists.
+func isBlocked(cert certificate, blockedCerts []string) bool {
+	// Special case for ca-certificates.crt
+	if cert.Name == "ca-certificates.crt" {
+		return true
+	}
+
+	// Check that the real underlying filepath to the certificate ends with .crt
+	if !strings.HasSuffix(cert.RealPath, ".crt") {
+		return true
+	}
+
+	// See if the name appears in the blockedCerts list
+	for _, blockedCert := range blockedCerts {
+		if cert.Name == blockedCert {
+			return true
+		}
+	}
+	return false
+}
+
+// TODO Should we support recursive directories inside the certs dir?
+func getCertObjects(basePath string) ([]certificate, error) {
+	certFiles, err := ioutil.ReadDir(basePath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read base certs: %v", err)
+	}
+
+	certsObjects := []certificate{}
```

Comment:
nitpick: I think this is more idiomatic/common
```suggestion
	var certsObjects []certificate
```